kore/src/Kore/Step/Transition.hs

```diff
@@ -161,6 +166,13 @@ mapRules f trans = do
     let results' = map (fmap (fmap f)) results
     scatter results'
 
+-- | Get the record of applied rules during an action.
+record :: TransitionT rule m a -> TransitionT rule m (a, Seq rule)
+record action = TransitionT $
```

Comment:
Could you take a look at adding some unit tests for this function? We already have some tests for one of the other functions in this module [here](https://github.com/kframework/kore/blob/master/kore/test/Test/Kore/Step/Transition.hs).